src/main/java/org/tron/p2p/P2pService.java

```diff
@@ -29,6 +29,8 @@ public void start(P2pConfig p2pConfig) {
     ChannelManager.init();
     DnsManager.init();
     log.info("P2p service started");
+
+    Runtime.getRuntime().addShutdownHook(new Thread(this::close));
   }
 
   public void close() {
```

Comment:
There's no flag here to prevent closing multiple times

Reply:
I hava added the flag isShutdown.